File: go/pivotIndex.go
func pivotIndex(nums []int) int {
  right := 0
  left := 0
  pivot := 0
  for _, num := range nums {
    right+=num
  }
  right-= nums[pivot]
  for left != right {
    pivot++
    if pivot >= len(nums) {
      return -1
    }
    right-=nums[pivot]
    left+=nums[pivot-1]
  }
  return pivot
}

File: javascript/pivotIndex.ts
function pivotIndex(nums: number[]): number {
    let left_sum = 0;
    let right_sum = 0;
    for (let i = nums.length - 1; i > 0; i--) {
        right_sum+=nums[i];
    }
    for (let i = 0; i < nums.length; i++) {
        if (left_sum === right_sum) {
            return i;
        } else {
            right_sum-=nums[i+1];
            left_sum+=nums[i];
        }
    }
    return -1
};
